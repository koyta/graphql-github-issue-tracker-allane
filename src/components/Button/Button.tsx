import { ComponentPropsWithoutRef, forwardRef } from "react";

import classNames from "classnames";

type Props = ComponentPropsWithoutRef<"button">;

export const Button = forwardRef<HTMLButtonElement, Props>(
  ({ type = "button", className, children, ...props }, ref) => {
    return (
      <button
        ref={ref}
        type={type}
        className={classNames(
          "cursor-pointer disabled:opacity-50 disabled:cursor-not-allowed disabled:hover:bg-blue-400 border border-blue-500 bg-blue-400 hover:bg-blue-500 transition-colors text-white p-2 rounded-md",
          className
        )}
        {...props}
      >
        {children}
      </button>
    );
  }
);
