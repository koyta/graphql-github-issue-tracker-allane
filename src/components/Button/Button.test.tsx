import { fireEvent, render } from "@testing-library/react";

import { Button } from "./Button";

describe("Button", () => {
  it("should render a button", () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container).toMatchSnapshot();
  });
  it("should render a button with a custom class", () => {
    const { container } = render(
      <Button className="custom-class">Click me</Button>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render a button with a custom type", () => {
    const { container } = render(<Button type="submit">Click me</Button>);
    expect(container).toMatchSnapshot();
  });
  it('should contain "Click me" text', () => {
    const { getByText } = render(<Button>Click me</Button>);
    expect(getByText("Click me")).toBeInTheDocument();
  });
  it("should be disabled", () => {
    const clickSpy = vi.fn();
    const { baseElement } = render(
      <Button disabled onClick={clickSpy}>
        Click me
      </Button>
    );
    fireEvent.click(baseElement);
    expect(clickSpy).not.toHaveBeenCalled();
  });
});
