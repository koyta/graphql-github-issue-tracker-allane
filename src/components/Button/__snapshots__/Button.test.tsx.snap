// Vitest Snapshot v1

exports[`Button > should render a button 1`] = `
<div>
  <button
    class="cursor-pointer disabled:opacity-50 disabled:cursor-not-allowed disabled:hover:bg-blue-400 border border-blue-500 bg-blue-400 hover:bg-blue-500 transition-colors text-white p-2 rounded-md"
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Button > should render a button with a custom class 1`] = `
<div>
  <button
    class="cursor-pointer disabled:opacity-50 disabled:cursor-not-allowed disabled:hover:bg-blue-400 border border-blue-500 bg-blue-400 hover:bg-blue-500 transition-colors text-white p-2 rounded-md custom-class"
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Button > should render a button with a custom type 1`] = `
<div>
  <button
    class="cursor-pointer disabled:opacity-50 disabled:cursor-not-allowed disabled:hover:bg-blue-400 border border-blue-500 bg-blue-400 hover:bg-blue-500 transition-colors text-white p-2 rounded-md"
    type="submit"
  >
    Click me
  </button>
</div>
`;
